EDX model documentation improvement

OliDG opened the pull request:
### Description of the change
The section about m.fit_background() is now placed right before the m.fit() as it should be. 
While subsequent m.fit_background() will actively change the background fit, the other model components are only updated when running m.fit() again.

### Progress of the PR
- [x] Change implemented (can be split into several points),
- [ ] update docstring (if appropriate),
- [x] update user guide (if appropriate),
- [ ] add an changelog entry in the `upcoming_changes` folder (see [`upcoming_changes/README.rst`](https://github.com/hyperspy/hyperspy/blob/RELEASE_next_minor/upcoming_changes/README.rst)),
- [ ] Check formatting changelog entry in the `readthedocs` doc build of this PR (link in github checks)
- [ ] add tests,
- [ ] ready for review.

### Minimal example of the bug fix or the new feature
# former doc sequence
m.fit() 
intensityA = m.get_line_intensity()
m.fit_background()    # modify the background but has no effect on the other components
intensityB = m.get_line_intensity()    # in this sequence instensityA = instensityB

# new doc sequence
m.fit_background()    # modify the background components and fix them
m.fit()     # fit the model components (expect the background) using the new background
intensityC = m.get_line_intensity()    # gives the correct line intensity based on the updated background

ericpre committed:
Merge remote-tracking branch 'upstream/RELEASE_next_patch' into RELEASE_next_minor

jlaehne committed:
typos useful information

ericpre committed:
Merge remote-tracking branch 'upstream/RELEASE_next_patch' into RELEASE_next_minor

ericpre committed:
Fix release version on RELEASE_next_minor

jlaehne committed:
Update python version in readthedocs.yml

 Update python version in readthedocs.yml

jlaehne committed:
seems to work only for python 3.8

ericpre committed:
Merge pull request #2980 from hyperspy/jlaehne-patch-1

 Update python version in readthedocs.yml

ericpre committed:
Merge remote-tracking branch 'upstream/RELEASE_next_patch' into RELEASE_next_minor

OliDG committed:
Update eds.rst

The section about m.fit_background() is now placed right before the m.fit() as it should be. 
While subsequent m.fit_background() will actively change the background fit, the other model components are only updated when running m.fit() again.

OliDG commented:
Hi, my bad, I realise that I forked the next_minor and try to commit to the next_patch, should have forked the next_patch instead but I didn't find a way.

jlaehne commented:
You need to `rebase` your local branch on next_patch and then push it.

ericpre commented on doc/user_guide/eds.rst:
```diff
@@ -577,22 +577,22 @@ background.
-
-    >>> m.fit()
-
 The background fitting can be improved with
```
With these changes, this wording needs to be changed too, as it is not "improved" from a previous fit anymore.

OliDG replied:
I made further changes as you suggested for improved consistency. How should I upload this second version? making a new branch? pulling a second request from my repository ?

OliDG replied:
I have committed the changes to my forked EDX_doc branch, I have renamed this branch to "hyperspy next_rebased" to reduce confusion, and rebased it on the "next_patch" branch. I tried to solved the conflicts and hope it can work!

OliDG committed:
Add files via upload

documentation about the EDS model update. m.background_fit() now appears before the m.fit(). Rephrasing for consistency.

ericpre commented:
@OliDG: it seems that there is something wrong. The easiest may be to make a new PR branching off `RELEASE_next_patch`.
